### PR TITLE
Use same handlers for sys policy and policies endpoints

### PR DESCRIPTION
**Pull request by calvn**

### CODE REVIEW

File: vault/logical_system.go [hunk 1]
```diff
@@ -2563,14 +2549,22 @@ func (b *SystemBackend) handlePoliciesList(policyType PolicyType) framework.Oper
 		case PolicyTypeACL:
 			// Add the special "root" policy if not egp
 			policies = append(policies, "root")
-			return logical.ListResponse(policies), nil
+			resp := logical.ListResponse(policies)
+
+			fmt.Println(req.Path)
```
Comment (jefferai) [inline vault/logical_system.go]: fmt?

File: vault/logical_system.go [hunk 2]
```diff
@@ -2563,14 +2549,22 @@ func (b *SystemBackend) handlePoliciesList(policyType PolicyType) framework.Oper
...
+			// If the request is from sys/policy/ we handle backwards compatibility
+			if strings.HasPrefix(req.Path, "policy") {
+				resp.Data["policies"] = resp.Data["keys"]
```
Comment (jefferai) [inline vault/logical_system.go]: Isn't this missing the code to add the root policy?

Comment (calvn) [reply vault/logical_system.go]: It's in https://github.com/hashicorp/vault/pull/4677/files/af11a3a72a3505362acdbce719470109d9802fc2#diff-a145099c8e3d917858619dcfb2ae09b9R2551

Comment (jefferai) [reply vault/logical_system.go]: oh nm I missed it

File: vault/logical_system.go (hunk 1 above)
Comment (calvn) [reply vault/logical_system.go]: Whoops

**Pull request merged**